### DISCUSSION TITLE
Some thoughts on editor icons

### DISCUSSION THREAD

**Discussion by passivestar** [godotengine/godot-proposals#10907] [Editor]
Godot has very clean and well-drawn icons. However some of them don't reflect very well the essence of what they're supposed to represent:

1. **Script files are using a cog icon.** While it's not completely off the mark, a cog is usually more strongly associated with config files than script files. It would make more sense to use it only for `.cfg` files

Take a look at flaticon.com top search results for "Config" and "Script":

| Config | Script |
|--------|--------|
| ![config](https://github.com/user-attachments/assets/b6819958-0c11-4ac8-b49a-c17c3343ef53) | ![script](https://github.com/user-attachments/assets/c6eeaba9-579a-4aa8-aad9-942ad1ec2ae8) |

Worth noting that the main screen button is using a scroll instead of a cog. It's probably best to use the same icon for both the script file and the button on top to make sure new users immediately recognize that the two relate to the same thing

<img width="296" alt="image" src="https://github.com/user-attachments/assets/5c2754e7-6f04-4cbd-9b87-3260432aa38c"><br>

Using a roll of papyrus to represent scripts might be a little bit dated. I want to emphasize that there's nothing wrong with using old metaphors for icons. It's ok to use a floppy disk for "Save" or old camera with film rolls for cameras. But it might be a good idea to go for a better consistency with other apps when it comes to the design language when possible. You just won't often see the papyrus scroll in software anymore. A good UI design is intentionally nonunique, predictable and boring

An unambiguous icon for a script file could be a text file icon with the language extension or logo overlayed on top, something like this (excuse my quick sketch):

<img src="https://github.com/user-attachments/assets/b394b63b-243b-4b8a-8a19-70d1c6e2f796" width=600/><br>

To quickly circle back to the "it's better to use the same icon everywhere" point, the scroll icon could be replaced with the text file icon with no extension overlay. It could be iterated on to make it more obvious that it's a code file, i.e round brackets

---

2. **Packed scene files are using a movie clapper icon.** Clapperboard is a good representation of the _word_ "Scene" (a movie scene to be exact), but it's not a good representation of _what a scene is_ in Godot. If you show this icon to somebody completely unfamiliar with Godot and ask them to guess what it is they'll likely say that it's a file that stores some kind of an animation or something related to making cinematic cutscenes

Admittedly it's not very easy to come up with an icon for such an abstract concept (a reusable/composable subtree of nodes). However I think a box is a better fit than a movie clapper because it doesn't clash semantically with animations/movies and it implies that it stores entities inside of it. In Godot you can **pack** branches by storing them on disc as a "PackedScene" resource and you can also **unpack** them (by making them local) which fits the box metaphor very well

| Clapperboard | Box |
|--------|--------|
| ![scene1](https://github.com/user-attachments/assets/234c4178-87f4-4f92-9df2-c3bce64d1904) | ![scene2](https://github.com/user-attachments/assets/bd4301ad-9611-46df-96a4-545d34e78101) |

---

3. **Settings button in the project manager is using a tools icon**.

Settings are usually much more strongly associated with a cog than a set of tools. Take a look at the flaticon search for "Settings" and the amount of cogs compared to wrenches:

<img src="https://github.com/user-attachments/assets/84b51638-f0a9-46a4-936c-746cc4b370e5" width=600/><br>

Here's a comparison:

| Tools | Cog |
|--------|--------|
| <img width="330" alt="settings1" src="https://github.com/user-attachments/assets/65b34889-275c-4599-986f-9632eeb49044"> | <img width="330" alt="settings2" src="https://github.com/user-attachments/assets/3ab59528-1010-404e-b3b4-df1b9c6a181a"> |

---

4. Finally, some resource icons like theme and bus layout are heavy on colors while most of the icons in Godot are monochrome (unless there's a need to show that they belong to the same base type). It would make sense to keep all of the resource icons monochrome for visual consistency unless there's a reason not to

Here are potential alternatives for those:

| Current | New |
|--------|--------|
| ![resources1](https://github.com/user-attachments/assets/9c4b743d-3bd0-4d92-9f81-26c749ac984a) | ![resources2](https://github.com/user-attachments/assets/d86b69f6-807b-4ad7-b48f-fe7972db67cc) |

---

In general, making the editor UI less surprising by choosing icons that better reflect associations that people already hold will make it easier for new users to get into Godot

**Comment by GuyUnger** [comment]
Definitely agree with the scripts/config icons.

I think a box for scenes is too generic, plus already used for resource base type. The clapperboard isn't perfect but once you learn the meaning makes for a pretty decent analogy.

I mix all types of files related to a game object in one folder, the rastafari colors for audio files helps a lot with readability, would be quite a regression for me if they become monochrome.

**Reply by passivestar** [reply]
> I mix all types of files related to a game object in one folder, the rastafari colors for audio files helps a lot with readability, would be quite a regression for me if they become monochrome

Hm, perhaps there's a version of this where there's some color accent in the icon but it's not so over the top, for example in this example the brush could be colored (but not the palette) and the level meter (but not the slider). There are some resource icons that have color like input events where it looks tasteful. But full on gradient over the whole icon seems a bit too much 😅

**Reply by GuyUnger** [reply]
you can also propose this in the redot fork, theyre probably eager to get rid of the rainbow gradient

**Reply by ettiSurreal** [reply]
Joking aside, 
I agree Theme-based resources should get their color changed, but primarily because they already share the rainbow gradient with Shader and Material based resources. I feel like Shaders fit a rainbow gradient way more since the point of Shaders is drawing pretty colors on your screen, though some Shader icons also need an overhaul imo.
I am not opposed to colored or gradient icons though, I think the Audio resource gradient looks nice. If anything I think more resource groups should be color-coded so they're more eye catching, like textures. But with that also a clear color language should also be defined so It doesn't become more chaotic than it already is.

**Comment by Mickeon** [comment]
I'd keep both icons for Scripts and Scenes the way they are, perhaps polish them up.
Not to say that these concepts don't exist elsewhere, but both are the bread and butter of Godot. They probably deserve a bit of uniqueness to them.

**Comment by ettiSurreal** [comment]
If I were to give Scenes a unique icon, it'd be one of a Branch of sorts. Scenes are preset _Branches_ of a Node _Tree_ so it just makes sense imo. Not sure how It'd work in practice but I might as well just throw this out here.

**Comment by azur-wolve** [comment]
agree with the script icon, always felt the cog a bit off
also with the settings being a cog instead of the tools

for scripts something like the papyrus with "{.gd}" | "{.cs}" | "{.json}" | ".ini" | ".cfg" 

the audio and other icons that are multi-color help distinguish them from the rest (those that are just about logic), would keep those

I really like the box for scenes, but its true that resources already use one
so what about having something like a node graph?
![image](https://github.com/user-attachments/assets/bfc12b73-75a2-4681-aaa7-edbcad95fe9d)


**Edited:**
What I would consider is to give a different icon for object distinct from resource; keep the box for resources.
And for the object.. maybe something like "Obj" / "O" in white? idk

Also for `PackedScene` the scene icon but boxed.

**Reply by passivestar** [reply]
> so what about having something like a node graph?

I thought about it but it's a bit too high-detail for such a common icon. Even if it only had 3 nodes. Also could potentially get confused with something graph-editor related

**Comment by Voxybuns** [comment]
Wholeheartedly agree on a monochrome version of those highly colorful resource icons, the current ones don't have great readability, especially the themeing related icons that use colors with very different luminances. That dark blue stripe in particular almost looks invisible against a dark mode background to my eyes, for instance.

They get even more muddied when inside a folder that has a color applied to it:
![image](https://github.com/user-attachments/assets/de7d40ec-281a-4440-bad1-78f9386f7054)

**Reply by passivestar** [reply]
Yeah like I mentioned under another comment they don't have to be full monochrome, they can be color-coded but less noisy like it was done for the input icons:

<img width="450" alt="image" src="https://github.com/user-attachments/assets/a4abf840-35ba-4b4d-9efa-0310565b2f5d">
